yt_dlp/extractor/youtube.py

```diff
@@ -3850,248 +3883,248 @@ def _music_reponsive_list_entry(self, renderer):
-        if not video_id:
             return
-        yield self._extract_video(renderer)
+        # Shelf may not contain shelf URL, fallback to extraction from content
+        yield from self.resolve_renderer(shelf_renderer, ctx=ctx)
 
-    def _video_entry(self, video_renderer):
+    def _video_entry(self, video_renderer, ctx=None):
         video_id = video_renderer.get('videoId')
         if video_id:
             return self._extract_video(video_renderer)
 
-    def _hashtag_tile_entry(self, hashtag_tile_renderer):
+    def _hashtag_tile_entry(self, hashtag_tile_renderer, ctx=None):
         url = urljoin('https://youtube.com', traverse_obj(
             hashtag_tile_renderer, ('onTapCommand', 'commandMetadata', 'webCommandMetadata', 'url')))
         if url:
             return self.url_result(
                 url, ie=YoutubeTabIE.ie_key(), title=self._get_text(hashtag_tile_renderer, 'hashtag'))
 
-    def _post_thread_entries(self, post_thread_renderer):
+    def _post_thread_entry(self, post_thread_renderer, ctx=None):
         post_renderer = try_get(
             post_thread_renderer, lambda x: x['post']['backstagePostRenderer'], dict)
         if not post_renderer:
             return
-        # video attachment
-        video_renderer = try_get(
-            post_renderer, lambda x: x['backstageAttachment']['videoRenderer'], dict) or {}
-        video_id = video_renderer.get('videoId')
-        if video_id:
-            entry = self._extract_video(video_renderer)
-            if entry:
-                yield entry
-        # playlist attachment
-        playlist_id = try_get(
-            post_renderer, lambda x: x['backstageAttachment']['playlistRenderer']['playlistId'], compat_str)
-        if playlist_id:
-            yield self.url_result(
-                'https://www.youtube.com/playlist?list=%s' % playlist_id,
-                ie=YoutubeTabIE.ie_key(), video_id=playlist_id)
+        # Attachment
+        yield from self.resolve_renderer(post_renderer.get('backstageAttachment'), ctx=ctx)
+
+        # TODO: sometimes may have an image but there is a link in the post
+        # e.g. https://www.youtube.com/post/Ugkx-Ic7tmOGjdYKqICrnxN3EuIQa2XOoYiP
         # inline video links
-        runs = try_get(post_renderer, lambda x: x['contentText']['runs'], list) or []
-        for run in runs:
-            if not isinstance(run, dict):
-                continue
-            ep_url = try_get(
-                run, lambda x: x['navigationEndpoint']['urlEndpoint']['url'], compat_str)
-            if not ep_url:
-                continue
-            if not YoutubeIE.suitable(ep_url):
-                continue
-            ep_video_id = YoutubeIE._match_id(ep_url)
-            if video_id == ep_video_id:
-                continue
-            yield self.url_result(ep_url, ie=YoutubeIE.ie_key(), video_id=ep_video_id)
+        # runs = try_get(post_renderer, lambda x: x['contentText']['runs'], list) or []
```

Comment:
Need to implement this. e.g. unavailable videos have an invalid videoRenderer - no id. So we need to extract from text